### ISSUE TITLE
ConstraintPanel and QCheckBox->text issue

### ISSUE THREAD

**Issue by rindeal** [pawelsalawa/sqlitestudio#3347]
`ConstraintPanel` and its derivatives like `TablePrimaryKeyPanel` use the `text` property of `QCheckBox` objects for reading of "valid data". This causes numerous issues throughout the program if `text` contains ampersands to generate `QShortcut`s, which on my machine is the case for at least all column names.

Take for example this code: https://github.com/pawelsalawa/sqlitestudio/blob/2266e39/SQLiteStudio3/guiSQLiteStudio/constraints/tableprimarykeypanel.cpp#L40-L83

Since `cb->text()` may contain ampersand somewhere in the column name, `createTable->getColumn(colName)` will fail and return `NULL`, thus https://github.com/pawelsalawa/sqlitestudio/blob/2266e39/SQLiteStudio3/guiSQLiteStudio/constraints/tableprimarykeypanel.cpp#L66-L73 will always get triggered and execution will never reach https://github.com/pawelsalawa/sqlitestudio/blob/2266e39/SQLiteStudio3/guiSQLiteStudio/constraints/tableprimarykeypanel.cpp#L82 .

**Comment by pawelsalawa** [comment]
Correct. There should be separation of model from view, meaning the logical name of a column should be kept in a separate (dynamic) property of a widget from the very creation of a widget and used from there. Thanks for pointing this out.

**Comment by rindeal** [comment]
So I traced the trigger of this problem to a KDE "feature" described here https://bugs.kde.org/show_bug.cgi?id=337491, which binds on to these widgets and replaces the `text` property with one including those ampersands. Nasty stuff.

**Comment by pawelsalawa** [comment]
Okay. I am testing on Kubuntu 16.04, but I'm using Qt installed from qt.io, not from the distribution. There it works correctly.

Unfortunately I'm unable to build SQLiteStudio with system's Qt5, because the scripting module of Qt seems to be unavailable, even though I have it installed

> $ sudo apt list libqt5script5
> List... Done
> libqt5script5/xenial,now 5.5.1+dfsg-2build1 amd64 [installed]

and I see it in:

> $ locate libQt5Script.so
> ...
> /usr/lib/x86_64-linux-gnu/libQt5Script.so.5
> /usr/lib/x86_64-linux-gnu/libQt5Script.so.5.5
> /usr/lib/x86_64-linux-gnu/libQt5Script.so.5.5.1

There is not libqt5script5-dev package at all.
I am using qmake from that very installation explicitly:

> ~/projects/sqlitestudio/scripts/linux$ ./compile.sh /usr/lib/x86_64-linux-gnu/qt5/bin/qmake

Maybe you have an idea, why it doesn't see the scripting module? I don't reallt get it. 

I will work on the issue in the mean time, but I'd like to be able to reproduce it.

**Comment by rindeal** [comment]
Well, `qtscript5-dev` is certainly there if you have `universe` repo enabled - https://packages.ubuntu.com/xenial/qtscript5-dev. I'll try to spin up a virtual machine to reproduce it.

**Comment by pawelsalawa** [comment]
Thanks. That did help. They have inconsistent naming. There is libqt**5**script5, but there is no "5" in between of qtscript5 package.

Anyway. It's compiling now. Thanks a lot for your help!

**Comment by pawelsalawa** [comment]
I got it reproduced!

In this issue in comments I will keep track on where this problem is observed.

@rindeal if you notice this problem in a place which I did not list here, please comment too, if you could. I would appreciate it very much!

My findings so far:

- All contraint panels that list columns
- Index dialog
- Populate table dialog
